### PR TITLE
Increase API timeout to handle longer endpoint build durations

### LINKED ISSUE #4995: Increase timeout on endpoint generations
Under some circumstances, the total duration of an endpoint build including waiting for the lock, doing the build and configuring the proxy can exceed the current timeout of 20 seconds, resulting in build failures.

Increase the timeout to 1 minute.

### LINKED ISSUE #4996: Increase API client timeout
As we are increasing the build timeout in #4995. The API client timeout of 30 seconds as implemented by go-swagger has to be increased as well.

### PR DESCRIPTION
**Pull request by nebril**
fixes #4995
fixes #4996

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4998)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

File: pkg/endpoint/bpf.go [hunk 1]
```diff
@@ -541,7 +541,7 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 	}
 
 	// Set up a context to wait for proxy completions.
-	completionCtx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	completionCtx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```
Comment (tgraf) [inline pkg/endpoint/bpf.go]: Please define a constant and add a comment to the constant to describe what the timeout it includes

Comment (nebril) [reply pkg/endpoint/bpf.go]: Done

**Comment by nebril** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by nebril** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by nebril** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hi! 

Quick question, for the CNI uses PUT /endpoint/ and the timeout for that is 25 seconds, should we increase that too?

https://github.com/cilium/cilium/blob/8d0f89c6bf0ec2698a0d9d6a4b2f6609792a793b/daemon/endpoint.go#L250

Regards

**Comment by tgraf** [comment]
> Quick question, for the CNI uses PUT /endpoint/ and the timeout for that is 25 seconds, should we increase that too?

Yes, that should be used as well. Ideally it uses the constant of this PR and adds a small buffer

**Comment by nebril** [comment]
I will go through the code and check where else should it be added

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by nebril** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
@nebril Are you still working on this? Looks good to go.

**Comment by tgraf** [comment]
Discussed offline, good to merge

**Comment by joestringer** [comment]
I suspect that this timeout didn't properly apply to endpoint creation events, see #5079 which is the next backport after #5041 (which included this fix and was already applied)

**Pull request merged**